pandas/tests/test_index.py

```diff
@@ -603,6 +603,157 @@ def test_shift(self):
         shifted.name = 'shifted'
         self.assertEqual(shifted.name, shifted.shift(1, 'D').name)
 
+    def test_union_naming_behavior(self):
+        #9965
+        idx_name_a = pd.Index([1,2,3], name='a')
+        idx_name_b = pd.Index([4,5,6], name='b')
+        idx2_name_a = pd.Index([2,9,8], name='a')
+
+        stridx_name_stra = pd.Index(['1','2'], name='stra')
+        stridx_name_a = pd.Index(['1','2'], name='a')
+
+        idx_name_none = pd.Index(['1','2'], name=None)
+
+        dateindex_name_a = pd.DatetimeIndex([datetime.today()], name='a')
+        dateindex_name_b = pd.DatetimeIndex([datetime.today()], name='b')
+        dateindex_name_None = pd.DatetimeIndex([datetime.today()], name=None)
+
+        python_array = [1,2,3]
+        numpy_array = np.array([1,2,3])
+
+        #index union index naming behavior
+        self.assertEqual(idx_name_a.union(idx_name_b).name, None)
```

Comment:
Can you change ``None`` comparisons to ``self.assertTrue(... is None)``? Sorry my test examples are incorrect. 
Or you can use ``check_index_equal`` to check names (see #9972).